### PR TITLE
fix(database): synchronize backups before snapshot publication

### LINKED ISSUE #3660: snapshots: publish manifests only after durable blob backup
A local snapshot manifest can become durable before the referenced blob backup has been fully synchronized. A restart can observe a committed manifest whose blob data was not durably completed.

Acceptance criteria:

- Complete and synchronize the blob backup before publishing the manifest.
- Synchronize the containing directory where required by the platform.
- Treat an interrupted backup as incomplete during discovery and restore.
- Add interruption and restart tests around each durability boundary.

### PR DESCRIPTION
**Pull request by wolf31o2**
Snapshot manifests could become durable before their backup contents. Synchronize both backup files, the snapshot directory, and any newly created ancestor entries before publishing the manifest. Flushes run after releasing the commit barrier.

Adds synchronization-failure and interrupted-process coverage for discovery and restore.

Fixes #3660.

### CODE REVIEW

File: database/lifecycle/snapshot.go [hunk 1]
```diff
@@ -225,12 +228,35 @@ func Snapshot(
...
+	metadataSyncErr := syncSnapshotFile(metadataFile)
+	metadataCloseErr := metadataFile.Close()
+	if err := errors.Join(metadataSyncErr, metadataCloseErr); err != nil {
+		return Manifest{}, fmt.Errorf("sync and close %q: %w", metadataPath, err)
+	}
+	// Persist the backup names before a manifest can make them discoverable.
+	if err := syncDir(dir); err != nil {
+		return Manifest{}, fmt.Errorf("sync snapshot directory: %w", err)
+	}
+	if err := syncDir(filepath.Dir(dir)); err != nil {
```
[the remark this answers is not in the document]
Comment (wolf31o2) [reply database/lifecycle/snapshot.go:275]: Snapshot now records missing destination ancestors before directory creation and synchronizes every new ancestor entry before publishing the manifest. Ancestor synchronization failures abort publication and remove the owned snapshot directory.

**Review by arepala-uml — APPROVED** [review]
Reviewed the fsync-ordering fix with filesystem-durability rigor. Confirmed the real pre-fix hazard: `Snapshot()` wrote the manifest immediately after backup goroutines finished, without ever fsyncing the backup file contents — a crash after that could leave a valid-looking manifest pointing at unflushed backup data.

- Hand-traced the fix's full sync sequence against a 3-level new-ancestor example and confirmed the exact set of directories synced is correct — nothing missing, nothing extra.
- The PR's own second commit exists because Cubic caught a real gap in the first (missing ancestor-directory fsync beyond the immediate parent) — confirmed fixed and tested at this head.
- Exceptional test quality: tests intercept sync calls to assert exact ordering, inject failures at each stage to confirm abort+cleanup, and one test actually re-execs the binary and calls `os.Exit()` at each durability checkpoint to simulate a genuine uncontrolled crash.
- `go build`/`vet`/`test -race`/`golangci-lint` all clean (86s test run). DATABASE.md accurate.

No merge-blocking issues found.
